File: src/main/java/io/bindernews/thegrackle/power/EmbodyFirePower.java
package io.bindernews.thegrackle.power;

import com.megacrit.cardcrawl.core.AbstractCreature;
import io.bindernews.thegrackle.GrackleMod;

public class EmbodyFirePower extends BasePower {
    public static final String POWER_ID = GrackleMod.makeId(EmbodyFirePower.class);

    public EmbodyFirePower(AbstractCreature owner, int amount) {
        super(POWER_ID);
        setOwnerAmount(owner, amount);
        loadRegion("flameBarrier");
        updateDescription();
    }

    @Override
    public void updateDescription() {
        description = formatDesc(0);
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/power/EmbodyFirePower.kt
package io.bindernews.thegrackle.power

import com.megacrit.cardcrawl.core.AbstractCreature
import io.bindernews.thegrackle.GrackleMod

class EmbodyFirePower(owner: AbstractCreature, amount: Int) : BasePower(POWER_ID) {
    init {
        setOwnerAmount(owner, amount)
        loadRegion("flameBarrier")
        updateDescription()
    }

    override fun updateDescription() {
        description = formatDesc(0)
    }

    companion object {
        @JvmField val POWER_ID = GrackleMod.makeId(EmbodyFirePower::class.java)
    }
}